tests/RobotFramework/tests/tedge_agent/workflows/custom_operation.robot

```diff
@@ -34,6 +34,14 @@ Trigger Device Restart
     ${expected_log}     Get File    ${CURDIR}/restart-command-expected.log
     Should Be Equal     ${actual_log}    ${expected_log}
 
+Trigger Device Restart Using A Sub-Command
+    [Documentation]    To detect if the device has been rebooted, a marker file is created in the /run directory
+        ...            which should be deleted when the device is restarted
+    Execute Command    sudo touch /run/boot1
```

Comment:
Seeing this, I'm wondering if we can update our `await-agent-restart` logic to do something similar (or introduce an `await-device-restart`)